ubuntudesign/tour.ubuntu.com
# Remove minified files from repository

I set out with the following goals:
- Remove minified source files from repository, leaving it in the "development" state
- Make sure `gulp` tasks can be run successfully on Ubuntu (it was failing)
- Run `gulp` in the build step - developers are not expected to run it

First I discovered that it was the `critical` task that failed on Ubuntu, and
for no obvious reason.

When I looked into this, I decided it was probably easier to just inline
_all_ the CSS and JS, as so much of it is critical anyway -
[a longer explanation here](8b1f7055a48b8fdea6955decb9646218005274d6).

Also, now that the `gulp` target is no longer part of the standard development
workflow, I decided to compress the `index.html` files inplace. However,
in local development, this could lead to losing work. So I renamed the
`default` target to `compress-all`, so hopefully you won't get devs doing
it by accident.

I probably will only run `compress-html` in the build-step, to speed it up,
leaving `compress-images` to be manually run and committed as and when we
feel like it. Then I won't include the `js` and `css` in the build.

PageSpeed now shows [85/100](https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Ftour.staging.ubuntu.com%2Fen%2F&tab=desktop) instead of [87/100](https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Ftour.ubuntu.com%2Fen%2F&tab=desktop). I've played around a lot, can't work out why. It's a weird beast and I don't trust it.

This has been successfully [deployed to staging](http://tour.staging.ubuntu.com/en/).

See also, the [code build](http://jenkins.demo.haus/job/tour.ubuntu.com/12/console) and the [staging release](https://website.ci.canonical.com/job/tour.ubuntu.com-staging/16/console).
# QA

``` bash
gulp compress-all
python -m SimpleHTTPServer 8000
```

Visit http://127.0.0.1:8000/en/ and check it still works.
